### ISSUE TITLE
feat: defer to prod

### ISSUE THREAD

**Issue by AntoineGiraudKeyrus** [dataform-co/dataform#1838]
Hi,

Could dataform add a [defer to prod](https://docs.getdbt.com/reference/node-selection/defer) option ?
i'd help to init our dev env' without having to clone/recreate all our parents tables when editing 1 child table !

resolving in compute, storage & time savings while developping

I'd help us a lot @Homeserve 👷‍♂️

DBT equivalent 🙏
![image](https://github.com/user-attachments/assets/433bed58-a112-4dcc-81a4-f9069052b0b6)

**Comment by HHammond** [comment]
Bumping this issue. 

As a newcomer to dataform from dbt this is something I would use every day.

Defer saves a lot of money in compute and makes development processes much much faster. It's a strong value add.

**Comment by andrebmallmann** [comment]
Hi @kolina 

Is this something that we should resolve on the CLI level or maybe its bigger then that?

A defer strager might work similar to DBT that use manifest files but seems a huge change. Any idea of implementation for that one?

**Comment by kolina** [comment]
Yeah, we don't have a state file in Dataform, implementing it is feasible, but will take some effort.

Alternatively we could support simplified version that just defers dependencies from a different environment, but this will require our compilation generation to remember dependency references in SQL so they can be overwritten during runtime (currently it produces static SQL with all references already resolved).
